File: scala/p9.scala
// Solution for Project Euler problem 9
// http://projecteuler.net/index.php?section=problems&id=9

import scala.Math._

println((1 to 1000).flatMap(x => (1 to x).
  map(y => List(x, y, (1000-x-y)))).
  filter(x => pow(x.apply(0), 2) + pow(x.apply(1), 2) == pow((1000-x.apply(0)-x.apply(1)), 2)).map(x => x.reduceLeft(_*_)))

File: python/p9.py
# Solution for Project Euler problem 9
# http://projecteuler.net/index.php?section=problems&id=9

for i in xrange(1, 50):
    for j in xrange(i+1, 50):
        a = 2*i*j
        b = j**2 - i**2
        c = i**2 + j**2
        if (a + b + c) == 1000:
            print a * b * c
            break

